IM_calculation/IM/im_calculation.py

```diff
@@ -130,22 +132,30 @@ def compute_measure_single(waveform, ims, comps, im_options, str_comps):
 
     if "PGV" in ims:
         value = intensity_measures.get_max_nd(velocities)
-        result[station_name]["PGV"] = array_to_dict(value, str_comps, "PGV", comps)
+        result[station_name]["PGV"] = array_to_dict(
+            value, comps_to_calculate, "PGV", comps_to_store
+        )
 
     if "PGA" in ims:
         value = intensity_measures.get_max_nd(accelerations)
-        result[station_name]["PGA"] = array_to_dict(value, str_comps, "PGA", comps)
+        result[station_name]["PGA"] = array_to_dict(
+            value, comps_to_calculate, "PGA", comps_to_store
+        )
 
     if "pSA" in ims:
-        value = intensity_measures.get_spectral_acceleration_nd(
-            accelerations, im_options["pSA"], waveform_acc.NT, DT
-        )
         # store a im type values into a dict {comp: np_array/single float}
         # Geometric is also calculated here
-        result[station_name]["pSA"] = (
-            im_options["pSA"],
-            (array_to_dict(value, str_comps, "pSA", comps)),
+        psa, u = intensity_measures.get_spectral_acceleration_nd(
+            accelerations, im_options["pSA"], waveform_acc.NT, DT
+        )
+        pSA_values = (
```

Comment:
A comment of the structure of this would be quite useful.

Reply:
It might be a better option to store 'u' as a separate result.

Reply:
u isn't a result, it's the spectral displacements which are an intermediate variable (updated for clarity)